community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/pipes/MergeIntoPipe.scala

```diff
@@ -69,7 +69,10 @@ case class MergeIntoPipe(source: Pipe,
           val relationships = findRelationships(fromNode, toNode, typeId)(state, row)
 
           if (relationships.isEmpty) {
-            val relationship = state.query.createRelationship(fromNode.getId, toNode.getId, typeId)
+            val relationship = if (dir == SemanticDirection.OUTGOING)
+                state.query.createRelationship(fromNode.getId, toNode.getId, typeId)
+              else
+                state.query.createRelationship(toNode.getId, fromNode.getId, typeId)
```

Comment:
Do we support `MERGE` with no direction on the pattern? If we do, I guess that would mean that `dir` here would be `SemanticDirection.BOTH`?

If that is the case I'd change that to `dir == SemanticDirection.INCOMING` and flip the two branches, so that the `BOTH` creates a relationship in the same direction as the pattern was written.